### DISCUSSION TITLE
Alarm Status

### DISCUSSION THREAD

**Discussion by cconst22313** [guerrerotook/securitas-direct-new-api#133] [Q&A] [answered]
Hi,
Thank you @guerrerotook for this huge effort!

I have one question: I am trying to use alarm state changes to trigger some actions (e.g. start my vacuum and close my blinds), but for some reason status is not being changed when I turn on or off the alarm on my phone app or at the door when I am leaving.

However if I turn the alarm on at my home assistant the automation works fine.

It seems like status changes are not being detected in my installation...

Is there something that I should look for?! 

BR
Carlos

**Comment by guerrerotook** [comment, marked as answer]
Hey @cconst22313 thanks for the message. This behavior has been documented and talked before. The reason for that is because the integration is on poll mode, meaning that it goes to the Securitas Direct API every 20 minutes (something that can be configured). And that is the reason why if you arm or disarm the alarm using the phone app instead of HA it will likely, in the worst time take up to 20 minutes. I also set the timer to update the status of the alarm to 20 minutes not to upset the API so they disable it or something. 

My recommendation is to always arm and disarm the alarm in HA to make sure that it trigger whatever you need. Eventually it will update and trigger whatever automation you have. 

Hope that clarify this!.

Luis.

**Reply by cconst22313** [reply]
Thanks for your response.

Even if It't not the one that I was looking for (I would be nice to react in real time but..) @guerrerotook thanks again for all of your effort! I wouldn't be capable of doing it better.
